### PR TITLE
MB-13754: Working client logging

### PR DESCRIPTION
**Pull request by ahobson**
## Summary

* Rework milmoveLogger to be compliant with @opentelemetry/api
* add componentDidCatch to SystemAdmin page to log errors

### Testing with console logging

1. Start the client: `make client_run`. Note that the `.envrc` changes ensure the client logs to the console
2. Start the server: `make server_run`.
3. Go to `http://milmovelocal:3000`
4. Open the dev tools and look at the console. You should see something like:
```
Operation users.showLoggedInUser failed: Error: Unauthorized (401)
(anonymous) @ consoleLogger.ts:49
```

### Testing sending logs to the server (like we will do in deployed environments)

1. Start the client: `REACT_APP_ERROR_LOGGING=otel make client_run`
2. Start the server: `SERVE_CLIENT_COLLECTOR=true make server_run`
3. Go to `http://milmovelocal:3000`
4. Look at the server logs. You should see something like:
```
2023-08-08T21:29:34.857Z	INFO	handlers/client_log_collector.go:48	client log upload stats	
{"git_branch": "adh-MB-13754-browser-logging", "git_commit": "9c073c46b1cf06a12aedc0ec08f325c7e5dac086", 
"source": "client_stats", "app": "my", "logEntryCount": 1, "droppedLogCount": 0, "failedSendCount": 0, 
"failedTimerCount": 0}
2023-08-08T21:29:34.857Z	INFO	handlers/client_log_collector.go:59	client log entry	
{"git_branch": "adh-MB-13754-browser-logging", "git_commit": "9c073c46b1cf06a12aedc0ec08f325c7e5dac086",
 "source": "client_log_entry", "app": "my", "logLevel": "error", "args": ["Operation users.showLoggedInUser failed: Error:
 Unauthorized (401)"]}
```

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### CODE REVIEW

File: src/utils/milmoveLog.js [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
-  INFO: 'INFO',
-  DEBUG: 'DEBUG',
-  LOG: 'LOG',
-};
+// use the diag logger from opentelemetry
+// This uses the logger configured via `diag.setLogger`, otherwise
+// calls to diag are noops
+import { diag, DiagConsoleLogger, DiagLogLevel } from '@opentelemetry/api';
 
-// milmoveLog: wrapper log to re-direct logging to something other than console.log/warn/error
```
Comment (YanZ777) [inline src/utils/milmoveLog.js:10]: **praise:** 🎉 Yay for removing code so that we don't have to have this annotation!

**Review by YanZ777 — COMMENTED** [review]
It looks good to me, but I'd want another pair of eyes on this since I am not one versed in logger stuff. 

I'd say the js stuff looks solid though.

**Comment by YanZ777** [comment]
Also, would this be useful to test in exp?

**Comment by rogeruiz** [comment]
@YanZ777 this is currently running in EXP now in this PR, #11208. Feel free to add commits if you want to further test the client-side errors.

File: yarn.lock
Comment (YanZ777) [inline yarn.lock:1]: **question:** Should there be corresponding changes inside of the package.json for these new dependencies?

Comment (ahobson) [reply yarn.lock:1]: @YanZ777 Thank you for checking. I think I see I added two packages to `package.json` and I see three additions to `yarn.lock`, one of which is a dependency of the other two. Are you seeing something else?

Comment (YanZ777) [reply yarn.lock:1]: Drat, I missed it and forgot there was a file filter! We're good here!

**Review by YanZ777 — APPROVED** [review]
Tested it out on exp and locally, and the changes look good to me.

**Pull request merged**